pandas/tests/io/pytables/test_store.py

```diff
@@ -79,33 +77,31 @@ def test_format_kwarg_in_constructor(self, setup_path):
 
         msg = "format is not a defined argument for HDFStore"
 
-        with ensure_clean_path(setup_path) as path:
+        with tm.ensure_clean(setup_path) as path:
             with pytest.raises(ValueError, match=msg):
                 HDFStore(path, format="table")
 
     def test_context(self, setup_path):
-        path = create_tempfile(setup_path)
-        try:
-            with HDFStore(path) as tbl:
-                raise ValueError("blah")
-        except ValueError:
-            pass
-        finally:
-            safe_remove(path)
-
-        try:
-            with HDFStore(path) as tbl:
-                tbl["a"] = tm.makeDataFrame()
-
-            with HDFStore(path) as tbl:
-                assert len(tbl) == 1
-                assert type(tbl["a"]) == DataFrame
-        finally:
-            safe_remove(path)
+        with tm.ensure_clean() as path:
+            try:
+                with HDFStore(path) as tbl:
```

Comment:
you can use contextlib.suppress here to make this much easier to read

Reply:
Wasn't sure what to do here

Reply:
I think the suggestion is to use `contextlib.suppress` to ignore the error.

e.g.
```python
In [1]: from contextlib import suppress

In [2]: with suppress(ZeroDivisionError):
   ...:     1/0
   ...:

In [3]: try:
   ...:     1/0
   ...: except ZeroDivisionError:
   ...:     pass
   ...:

```

Reply:
So the suggestion is to use contextlib.suppress in order to change the following lines of code. Im not sure exactly how to do so given that the code being tested here is a context manager itself

```
            try:
                with HDFStore(path) as tbl:
                    raise ValueError("blah")
            except ValueError:
                pass
```

Reply:
hmm ok then leave this i guess